### PR TITLE
CAPK support to install docker

### LINKED ISSUE #1434: CAPK supports to install docker
### Your current KubeKey version

latest

### Describe this feature

At present, cluster-api-kubekey-provider(CAPK) only supports to install `containerd`. We need to support installing `docker`.

### Describe the solution you'd like

Add a `dockerService` in the capk `pkg/service/containermanager`.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by arugal**
### What type of PR is this?

/kind feature

### Which issue(s) this PR fixes:

Fixes #1434

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/service/containermanager/docker.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		return err
+	}
+	return nil
+}
+
+func (d *DockerService) Install() error {
+	if err := d.installDocker(); err != nil {
+		return err
+	}
+	// TODO install https://github.com/Mirantis/cri-dockerd
```
Comment (arugal) [inline pkg/service/containermanager/docker.go:129]: @24sama Do we need to install [cri-dockerd](https://github.com/Mirantis/cri-dockerd) ?

**Review by 24sama — COMMENTED** [review]
Thanks for your contributions!

File: pkg/service/containermanager/docker.go (hunk 1 above)
Comment (arugal) [reply pkg/service/containermanager/docker.go:129]: > I think capkk can determine if need to install cri-dockerd based on the Kubernetes version

Yes, we can judge the kuberentes version

> Do you think we need to add some fields in the [api](https://github.com/kubesphere/kubekey/blob/master/api/v1beta1/containermanager_types.go) or just hard-code its version?

Yes, and also need a chinese download link

**Comment by arugal** [comment]
### 1.23.7
image

### 1.24.3
image

For the latest code, deployed `cri-dockerd` throughout when `CRI` type is docker, and it worked fine in 1.23 and 1.24. Please review @24sama

**Comment by 24sama** [comment]
Looks good to me! BTW, you can use `make docker-build-e2e` and `make test-e2e` to do a quick-start e2e test.

The e2e conf is here, you can modify it based on your environment:
https://github.com/kubesphere/kubekey/blob/70bc1ab4bad85fae727d2cbfdacf0f824b2b8c71/test/e2e/config/e2e_conf.yaml#L105

**Comment by 24sama** [comment]
Thanks again! @arugal 
/lgtm
/approve

**Comment by 24sama** [comment]
@all-contributors please add @arugal for code

**Pull request merged**